### PR TITLE
WIP: Add traffic router component to cdn-in-a-box

### PR DESCRIPTION
**Pull request by JBevillC**
#### What does this PR do?

This PR will add the traffic router component to the cdn-in-a-box docker-compose project

TODO: 
 - Generate SSL certificates and add them to the JDK cacerts file
 - Turn on the enroller registration once that PR gets merged.
 - Tail the base tomcat log files.
 - Java8 version may be too old.

WORKAROUNDS:
 - Traffic Router RPM has sudo commands embedded within the scriptlet, so rpm2cpio is used

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [X] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [X] Other _CDN-In-A-Box_____

#### What is the best way to verify this PR?

docker-compose build trafficrouter
docker-compose up trafficrouter

#### Check all that apply

- [ ] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [X] This PR does *NOT* fix a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2354/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2360/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2361/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2362/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2363/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2364/
Test PASSed.

File: infrastructure/cdn-in-a-box/traffic_router/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+############################################################
+# Dockerfile to build Traffic Router 3.0 
+# Based on CentOS 7.x
+############################################################
```
Comment (dangogh) [inline infrastructure/cdn-in-a-box/traffic_router/Dockerfile]: not for building traffic router -- actually,  I'd remove this comment completely -- the FROM tells what it's based on

File: infrastructure/cdn-in-a-box/traffic_router/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+FROM centos:7
+MAINTAINER Dan Kirkwood
```
Comment (dangogh) [inline infrastructure/cdn-in-a-box/traffic_router/Dockerfile]: remove the MAINTAINER line

File: infrastructure/cdn-in-a-box/traffic_router/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+### Common for all sub-component builds
+RUN	yum -y install epel-release git rpm-build net-tools iproute nc wget tar unzip \
+          perl-JSON perl-WWWCurl which tcpdump autoconf automake gcc gcc-c++ \
+          apr apr-devel java-1.8.0-openjdk java-1.8.0-openjdk-devel make \
+					openssl openssl-devel && \
+	  yum -y clean all
+
+WORKDIR /opt
+RUN curl -L -b "oraclelicense=a" http://download.oracle.com/otn-pub/java/jdk/8u131-b11/d54c1d3a095b4ff2b6607d096fa80163/jdk-8u131-linux-x64.tar.gz | tar xzf - && \
+    curl -L https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.3.9/apache-maven-3.3.9-bin.tar.gz | tar xzf -
```
Comment (dangogh) [inline infrastructure/cdn-in-a-box/traffic_router/Dockerfile]: may want to provide these paths as ARGS with these defaults to make it easier to override

File: infrastructure/cdn-in-a-box/traffic_router/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+RUN alternatives --install /usr/bin/java java /opt/java/bin/java 2 && \
+    alternatives --install /usr/bin/jar jar /opt/java/bin/jar 2 && \
+    alternatives --install /usr/bin/javac javac /opt/java/bin/javac 2 && \
+    alternatives --set java /opt/java/bin/java && \
+    alternatives --set jar /opt/java/bin/jar && \
+    alternatives --set javac /opt/java/bin/javac && \
+    alternatives --install /usr/bin/mvn mvn /opt/maven/bin/mvn 1
+
+# Install Tomcat and Tomcat Native 
+RUN cd / && rpm2cpio /traffic_router.rpm | cpio -ivd && \
```
Comment (dangogh) [inline infrastructure/cdn-in-a-box/traffic_router/Dockerfile]: add a TODO: change to yum install when....

File: infrastructure/cdn-in-a-box/traffic_router/to-access.sh [hunk 5]
```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
```
Comment (dangogh) [inline infrastructure/cdn-in-a-box/traffic_router/to-access.sh]: don't want to have multiple copies of to-access.sh -- that's why all the containers have context set to the cdn-in-a-box level -- remove this and `ADD traffic_ops/to-access.sh` in the dockerfile

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2373/
Test PASSed.

**Comment by dangogh** [comment]
the implementation was included in a previously merged PR

**Pull request closed without merging**